scheduler/pkg/kafka/pipeline/broadcaster.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright (c) 2024 Seldon Technologies Ltd.
+
+Use of this software is governed by
+(1) the license included in the LICENSE file or
+(2) if the license included in the LICENSE file is the Business Source License 1.1,
+the Change License after the Change Date as each is defined in accordance with the LICENSE file.
+*/
+
+package pipeline
+
+import "sync"
+
+type Broadcaster struct {
+	mu    sync.Mutex
+	ready chan struct{}
+}
+
+func NewBroadcaster() *Broadcaster {
+	return &Broadcaster{}
+}
+
+func (b *Broadcaster) Subscribe() <-chan struct{} {
```

Comment:
First try to read lock using `RLock` & `RUnlock`. Only if the channel is not ready, then create the channel. This would allow multiple subscribers to subscribe simultaneously if the channel is already initialized.